### PR TITLE
User roles

### PR DESCRIPTION
**Pull request by stephenhammond**
- first user is assigned as Presenter
- subsequent logged in users are designated as "viewers"
- Presenters can promote viewers to presenters
- Presenters can demote other presenters to become viewers
- anonymous users can see list of other logged in users
- logged in user count is displayed
- control fields are shown based on user role

### CODE REVIEW

File: client/application/conf.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<template name="conf">
+<!-- http://www.rtcmulticonnection.org/docs/getting-started/	 -->
+	<button id="openNewSessionButton">Open New Room</button>
+</template>
```
Comment (nuen) [inline client/application/conf.html]: You can go ahead and delete this file.

File: client/application/conf.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	// setup signaling channel
+	connection.connect("commonchannel");
+
+	// open new session
+	document.querySelector('#openNewSessionButton').onclick = function() {
+	    connection.open("commonchannel");
+	};
+
+})
```
Comment (nuen) [inline client/application/conf.js]: You can also delete this file.

File: client/application/editor.html [hunk 3]
```diff
@@ -3,7 +3,11 @@
   {{#if isViewingBoard}}
     {{> blackboard}}
   {{else}}
-    {{> sharejsAce docid=currentTab id="editor"}}
+	<div class="syntax-container"> 
+		Syntax options:
+		{{> mode}}
+	</div>
+    {{> sharejsAce onRender=config onConnect=setMode docid=currentTab id="editor"}}
```
Comment (nuen) [inline client/application/editor.html]: @mikegazdag Will this affect the syntax highlighting code you were working on?

**Comment by mikegazdag** [comment]
Hmm..hopefully this didn't break anything :p

**Pull request merged**